content/en/docs/Advanced/revoking-secureboot-access.md

```diff
@@ -0,0 +1,300 @@
+the UEFI databases have been altered, e.g. by enrolling a new key.
+
+And although this makes sense, security wise, it's exactly what we are trying to achieve here. We
+want to enroll a certificate in `dbx`. This would change the value of PCR 7 thus
+decryption of the disk partitions will no longer be possible after reboot.
+
+The method we use to overcome this issue is this:
+- We "unbind" decryption from PCR 7, binding only to PCR 11 indirectly.
+- We enroll the blacklisted cert in dbx.
+- We reboot to the upgraded system.
+- We bind decryption again both to PCR 11 and 7 (as before).
+
+When we rebind to PCR 7, the register has the new value which includes the cert in dbx.
+
+### Steps
+
+With the above now clarified, here are the steps to revoke the `db.pem` certificate from user space in Kairos.
+
+#### Generate a new db cert
+
+Since it's only the db key that we are trying to replace, we can start by copying
+the original directory generated by `enki genkey` and remove the old db files.
+
+```bash
+cp keys new-keys
+rm new-keys/db.*
+```
+
+Now create the new db files:
+
+```bash
+uuidgen --random > GUID.txt
+openssl req -newkey rsa:4096 -nodes -keyout db.key -new -x509 -sha256 -days 3650 -subj "/CN=NewKairosDB/" -out db.crt
+openssl x509 -outform DER -in db.crt -out db.cer
+cert-to-efi-sig-list -g "$(< GUID.txt)" db.crt db.esl
+sign-efi-sig-list -g "$(< GUID.txt)" -k KEK.key -c KEK.pem db db.esl db.auth
+```
+
+If you are planning to use `enki build-uki` command to prepare the new OS image,
+create the following formats too or the command might complain if they are missing:
+
+```bash
+$ openssl x509 -outform der -in db.crt -out db.der
+$ openssl x509 -inform DER -outform PEM -in db.der -out db.pem
+```
+
+#### Sign the new image with the new certificate
+
+The `new-keys` directory created above can be used to prepare the new image with `enki`. E.g.
+
+```bash
+docker run --rm -v $PWD/unpacked:/unpacked \
+  -v $PWD/build:/result \
+  -v $PWD/new-keys/:/keys \
+  quay.io/kairos/osbuilder-tools:v0.201.0 \
+  build-uki dir:/unpacked \
+  --output-dir /result \
+  --keys /keys \
+  --output-type container \
+  --boot-branding "KairosNewUKI"
+```
+
+#### Enroll the new certificate in db
+
+From withing Kairos (still booted in the old image), enroll the new db key:
+
+```bash
+sudo chattr -i /sys/firmware/efi/efivars/{PK,KEK,db}*
+efi-updatevar -f db.auth db
+```
+
+{{% alert title="Warning" %}}
+You will need to `efitools` installed on the Kairos image for this commands to work!
+{{% /alert %}}
+
+#### Enroll the old certificate in dbx
+
+As describe earlier in this document, you should generate a `db-dbx.auth` file
+from the old db key. This should now be enrolled in dbx. In user space again
+(booted in the old Kairos image):
+
+```bash
+sudo efi-updatevar -f db-dbx.auth dbx
+```
+
+
+#### Upgrade to the new image
+
+```
+kairos-agent upgrade --source oci:myimage
+```
+
+#### Unbind PCR 7
```

Comment:
I think the ordering is wrong here no?  You mentioned above

- We "unbind" decryption from PCR 7, binding only to PCR 11 indirectly.
- We enroll the blacklisted cert in dbx.
- We reboot to the upgraded system.
- We bind decryption again both to PCR 11 and 7 (as before).

But here we are  doing it on a different order in the docs. Would it make sense to have it in the correct order?

Reply:
Good point. Actually the order between enrolling to dbx and upgrading doesn't matter. We just can't reboot before we unbind from PCR 7. But I'll re-order to avoid confusing people.

Reply:
re-ordered